scala2.13/pom.xml

```diff
@@ -1492,10 +1558,10 @@ This will force full Scala code rebuild in downstream modules.
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>3.3.0</version>
+                <version>3.5.0</version>
                 <executions>
                   <execution>
-                    <id>enforce-maven</id>
+                    <id>default-cli</id>
```

Comment:
changing excution id here caused the buildver fail to get correct var from maven life cycle

```
Unexpected buildver value 311 for a Scala 2.13 build, only Apache Spark versions 3.3.0 (330) and higher are supported, no vendor builds such as 330db
```

Reply:
Yes I don't think the enforcer was actually working before changing the id. We need this value from command line to run the enforcer

Reply:
OK then you are changing the expected behavior of running mvn run with default builder.

This means we cannot simply mvn verify in scala213. if this is expected then we must doc this for all occurences.

Reply:
and require to update all corresponding scripts like
https://github.com/NVIDIA/spark-rapids/blob/branch-24.08/jenkins/spark-premerge-build.sh#L209

Reply:
I ran below in scala2.13 folder,
```
mvn help:evaluate -Dexpression=buildver  -q -DforceStdout
```
only showed 311 in this PR, 
for latest upstream, it could print out correct 330.

please make sure it could print out the correct builder for default profile locally

Reply:
@razajafri  @gerashegalov  Please note I've merge PR dropping 31x shims: https://github.com/NVIDIA/spark-rapids/pull/11159

In this PR there're some files' change which are conflicted[modified or deleted in MR11159], can you help sync those files with the TOT source on the upstream branch-24.08? Thanks!

Reply:
> I ran below in scala2.13 folder,
> 
> ```
> mvn help:evaluate -Dexpression=buildver  -q -DforceStdout
> ```
> 
> only showed 311 in this PR, for latest upstream, it could print out correct 330.
> 
> please make sure it could print out the correct builder for default profile locally
> 
> ```
> # try check
> mvn help:active-profiles
> ```
> 
> upstream:
> 
> ```
> Active Profiles for Project 'com.nvidia:rapids-4-spark-parent_2.13:pom:24.08.0-SNAPSHOT':
> 
> The following profiles are active:
> 
>  - release330 (source: com.nvidia:rapids-4-spark-parent_2.13:24.08.0-SNAPSHOT)
> ```
> 
> vs this PR:
> 
> ```
> Active Profiles for Project 'com.nvidia:rapids-4-spark-parent_2.13:pom:24.08.0-SNAPSHOT':
> 
> The following profiles are active:
> 
>  - enforce-cloudera-jdk-version (source: com.nvidia:rapids-4-spark-parent_2.13:24.08.0-SNAPSHOT)
> ```
> 
> Maven is prioritizing it over the default release activation (comment out this profile could result correct builder). I think you could just add the enforcer to cdh profiles directly if it's just a JDK version check


Looks like when building Scala2.13 + JDK17, it enables `enforce-cloudera-jdk-version` with condition <jdk>[17,)</jdk>

Then the <activeByDefault> 330 is disabled unless we manually add -Dbuildver=330, otherwise buildver falls back to the initial value of 311

Reply:
I created the `enforce-cloudera-jdk-version` to avoid code duplication as maven doesn't allow profiles inheritance. I have updated cloudera profiles now

Reply:
This is what I see now 
```
mvn help:active-profiles
```
in the root project dir 
```
Active Profiles for Project 'com.nvidia:rapids-4-spark-parent_2.12:pom:24.08.0-SNAPSHOT':

The following profiles are active:

 - release320 (source: com.nvidia:rapids-4-spark-parent_2.12:24.08.0-SNAPSHOT)
```
This is what I see in scala2.13 folder 
```
Active Profiles for Project 'com.nvidia:rapids-4-spark-parent_2.13:pom:24.08.0-SNAPSHOT':

The following profiles are active:

 - release330 (source: com.nvidia:rapids-4-spark-parent_2.13:24.08.0-SNAPSHOT)
```